lightning/src/routing/router.rs

```diff
@@ -829,6 +831,20 @@ where L::Target: Logger {
 	// - when we want to stop looking for new paths.
 	let mut already_collected_value_msat = 0;
 
+	// We only consider paths shorter than our maximum length estimate.
+	// In the legacy onion format, the maximum number of hops used to be a fixed value of 20.
+	// However, in the TLV onion format, there is no fixed maximum length, but the `hop_payloads`
+	// field is always 1300 bytes. As the `tlv_payload` for each hop may vary in length, we have to
+	// estimate how many hops the route may have so that it actually fits the `hop_payloads` field.
+	//
+	// We estimate 3+32 (payload length and HMAC) + 1+8 (amt_to_forward) + 1+4 (outgoing_cltv_value) +
```

Comment:
Aaah, I was wondering about this, then I misunderstood the spec there. BOLT #1 states: "The type field indicates how to interpret the payload field. The format for each individual type is defined by a specification in this repository." (cf. https://github.com/lightning/bolts/blob/master/01-messaging.md#lightning-message-format). This together with the fact that the `tlv_payload` format is [defined](https://github.com/lightning/bolts/blob/master/04-onion-routing.md#tlv_payload-format) for example as as
```
type: 2 (amt_to_forward)
data:
    [tu64:amt_to_forward]
```
that is, omitting the length field threw me off.
 
So I thought, depending on the given type, the length value might or might not be present. In this case, since length is implied in the `tu64` encoding, it could be omitted. But it seems that since `tlv_payload` is a `tlv_stream` **all** entries are `tlv_record`s and hence the length field is implied to be present between `type` and `data`?

On the other point: yes, I'm aware that `tu64`/`tu32` etc. might not take the full 8/4 etc. bytes, but chose to assume it to get a worst-case/conservative estimate.

Reply:
Btw, when I run the math for 3 bytes more per hop, I arrive at `19.7868`, i.e., still `20` rounded or `19` rounded down. 

Another question would be whether we want to account for the possible presence of some length `payment_metadata`. I'd say probably not, because we do not want to risk discarding feasible paths here for onions that that do not carry such metadata.

Reply:
> Sure, lets use 19, then. Worst case we ignore a 20-hop path, but honestly such a path is unlike to work anyway, and it reduces the chance of hitting [#1476 (comment)](https://github.com/lightningdevkit/rust-lightning/pull/1476#discussion_r871633632)

Alright, done with https://github.com/lightningdevkit/rust-lightning/pull/1476/commits/b98e98e558810dab712414dc8ae5b358907ff06b.